py3status/formatter.py

```diff
@@ -296,10 +296,35 @@ class Placeholder:
     Class representing a {placeholder}
     """
 
+    sanitized_regex_cache = {}
+    sanitized_string_cache = {}
+
     def __init__(self, key, format):
         self.key = key
         self.format = format
 
+    def _sanitize(self, string):
+        """
+        Remove redundant metadata from string and return it
+        """
+        try:
+            return self.sanitized_string_cache[string]
+        except KeyError:
+            cache_key = string
```

Comment:
Is this needed? The composites may could be cached. Maybe regex cache only... Something to look into.